src/internal/connector/sharepoint/libraries.go

```diff
@@ -0,0 +1,65 @@
+package sharepoint
+
+import (
+	"context"
+
+	"github.com/alcionai/corso/src/internal/connector/graph"
+	"github.com/alcionai/corso/src/internal/connector/onedrive"
+	"github.com/alcionai/corso/src/internal/connector/support"
+	"github.com/alcionai/corso/src/internal/data"
+	"github.com/alcionai/corso/src/pkg/logger"
+	"github.com/alcionai/corso/src/pkg/selectors"
+)
+
+func CollectLibraries(
```

Comment:
I'm confused by the purpose of this function. It seems like the calling function (`createSharePointCollections`) already iterates over all the sites in the scope. Won't that just result in fetching everything 2x? Also, since this is pretty much a mirror image of getting OneDrive data, why is it not in the `connector/data_collections.go` file so it is easier to see that it should be kept mostly in sync with the OneDrive call?

Reply:
> Won't that just result in fetching everything 2x?

Nope.  Though I can see how the code makes that confusing.  At this point we've separated the scopes into `DiscreteScopes(SiteIDs...)`, and so createSharePointCollections iteration of scope site IDs should only produce a single site.  Still, feels weird to range over that twice, so I'll just send in a single siteID instead.

> why is it not in the data_collections.go

Well, now that you mention it, I realized I went halfway towards the solution I actually wanted.  It felt a bit strange to me that we have a package per service, but also a graph connector file dedicated to per-service processing.  I wanted to break that pattern with sharepoint, and apparently forgot about that goal through the implementation, leaving half of the code in data_collections anyway.

sharepoint/libraries.go has been swapped for sharepoint/data_collections.go, and the handling moved into that space.  If we're concerned about code re-use (I'm not particularly worried here, but to address it), the ideal would be to have onedrive make the same refactor, and from there to choose an appropriate point for sharepoint to turn to onedrive for the creation/population of collections.